### PR TITLE
Add timing for query parsing to GraphQLController

### PR DESCRIPTION
**Pull request by bryan-coursera**
This will let us see if there are random slowdowns in query parsing. If not, we'll instrument more!

PTAL @yifan-coursera or @mkovacs-coursera

### CODE REVIEW

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala [hunk 1]
```diff
@@ -39,14 +39,27 @@ class MetricsCollectionMiddleware(
...
 class LoggingMetricsCollector extends GraphQLMetricsCollector with StrictLogging {
   def markFieldError(fieldName: String): Unit = {
     logger.info(s"Error when loading field $fieldName")
   }
+
+  def timeQueryParsing[A](f: => A): A = {
+    val before = System.currentTimeMillis()
+    val res = f
+    val after = System.currentTimeMillis()
+    logger.info(s"Parsed query in ${after - before}ms")
```
Comment (keith-coursera) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala]: This message would be much more useful if you passed in and logged `operation` here.  Maybe some graphQL queries are slower than others.

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala [hunk 2]
```diff
@@ -118,7 +118,10 @@ class GraphQLController @Inject() (
       variables: JsObject,
       operation: Option[String]): Future[OutgoingQuery] = {
     Future {
-      QueryParser.parse(query) match {
+      val parsedQuery = metricsCollector.timeQueryParsing {
+        QueryParser.parse(query)
+      }
```
Comment (keith-coursera) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala]: This is a start.  Do you think the later assignments to `incomingQuery`, `filterFn` and the call to `filterFn(incomingQuery)` are worth tracking, too?

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala (hunk 1 above)
Comment (keith-coursera) [reply naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala]: Another note about this.  This is going to probably cause a lot of logging.  Consider logging at `debug` level, then we turn it on while triaging?  I'm not sure how you guys normally do this.

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala (hunk 2 above)
Comment (bryan-coursera) [reply naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala]: If this doesn't give us any actionable data, we can add some more tracking in. I'm pretty sure those are all very cheap operations, so I don't want to over-collect data. But it's easy enough to add it in later.

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala (hunk 1 above)
Comment (bryan-coursera) [reply naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala]: This MetricsCollector isn't designed to be used in production (we override this with a DatadogMetricsCollector in our repo), so this is mostly just used for debugging in the `example` project

Comment (bryan-coursera) [reply naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala]: ah, but I could tag the metric with operation name. good point

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala (hunk 2 above)
Comment (keith-coursera) [reply naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala]: agreed

**Pull request merged**